### PR TITLE
Added the local training baseline with clients' local early-stoppers

### PR DESCRIPTION
**Pull request by yxdyc**
- added the local training baseline
- enabled each client has its own early-stopper

### CODE REVIEW

File: federatedscope/core/monitors/monitor.py [hunk 1]
```diff
@@ -176,3 +176,98 @@ def calc_blocal_dissim(last_model, local_updated_models):
             b_local_dissimilarity[k] = np.sqrt(
                 avg_gnorms[k].item() / torch.sum(global_grads[k]**2).item())
         return b_local_dissimilarity
+
+
+def update_best_result(best_results,
```
Comment (rayrayraykk) [inline federatedscope/core/monitors/monitor.py:181]: Please return `best_results`, or the value won't be changed.

Comment (rayrayraykk) [inline federatedscope/core/monitors/monitor.py:181]: Please return `best_results`, or the value won't be changed.

**Review by rayrayraykk — COMMENTED** [review]
Please fix the bug, thanks!

**Review by DavdGao — APPROVED** [review]
looks good to me

**Pull request merged**